test/modules/auxiliary/test/sqli_test.rb

```diff
@@ -112,11 +124,32 @@ def perform_sqli(sqli)
 
   def run
     case datastore['SQLI_TYPE']
-    when 0
+    when 'Regular'
+      @dbms = case datastore['DBMS']
+        when 'MariaDB' then Msf::Exploit::SQLi::MySQLi::Common
+        when 'PostgreSQL' then Msf::Exploit::SQLi::PostgreSQLi::Common
+        when 'Sqlite' then Msf::Exploit::SQLi::SQLitei::Common
+        when 'MSSQL' then Msf::Exploit::SQLi::Mssqli::Common
+        else print_bad("Unsupported target")
+      end
       reflected
-    when 1
+    when 'BooleanBlind'
+      @dbms = case datastore['DBMS']
+        when 'MariaDB' then Msf::Exploit::SQLi::MySQLi::BooleanBasedBlind
+        when 'PostgreSQL' then Msf::Exploit::SQLi::PostgreSQLi::BooleanBasedBlind
+        when 'Sqlite' then Msf::Exploit::SQLi::SQLitei::BooleanBasedBlind
+        when 'MSSQL' then Msf::Exploit::SQLi::Mssqli::BooleanBasedBlind
+        else print_bad("Unsupported target")
+      end
       boolean_blind
-    when 2
+    when 'TimeBlind'
+      @dbms = case datastore['DBMS']
+        when 'MariaDB' then Msf::Exploit::SQLi::MySQLi::TimeBasedBlind
+        when 'PostgreSQL' then Msf::Exploit::SQLi::PostgreSQLi::TimeBasedBlind
+        when 'Sqlite' then Msf::Exploit::SQLi::SQLitei::TimeBasedBlind
+        when 'MSSQL' then Msf::Exploit::SQLi::Mssqli::TimeBasedBlind
+        else print_bad("Unsupported target")
+      end
       time_blind
     else
       print_bad('Unsupported SQLI_TYPE')
```

Comment:
Same here, I believe `SQLI_TYPE` option is an enum and this should not be necessary.